### PR TITLE
Use Icon.x() for select clear indicator

### PR DESCRIPTION
**Pull request by amcclain**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by amcclain** [comment]
Hi @TomTirapani - can you take a look at this whenever you have a few spare minutes and let me know if you think it's an improvement?

If you add `enableClear: true` to the Toolbox > Forms > FormModel text and select fields it lets you compare the two and see how I was trying to match them better.

The new X icon is at once more consistent but also undeniably larger. Made me wonder if we need/want to use the large font size for the BP controls. but I really didn't want to revisit that now.

I don't feel strongly about this at all but it was bugging me on an app form I was building how different the clear indicators were across the input types.

File: desktop/cmp/input/Select.js [hunk 1]
```diff
@@ -515,6 +516,18 @@ export class Select extends HoistInput {
             () => Icon.chevronDown({className: 'xh-select__indicator'});
     }
 
+    // As per example @ https://react-select.com/components#replaceable-components
```
Comment (lbwexler) [inline desktop/cmp/input/Select.js]: just curious about the asymmetry with our custom dropdown indicator above.    Are we potentially missing something there with respect to refs, and additional props?

Comment (amcclain) [reply desktop/cmp/input/Select.js]: I had the same question - I can tell you the clear indicator required the innerProps to be spread on the div as that's what wired up the correct / provided click handler.

**Comment by lbwexler** [comment]
I think its fine -- looks like the Dropdown example they show is actually
simpler and does not need that ref.

Will merge now

**Comment by TomTirapani** [comment]
Looks great, much better than the inconsistent tiny one before :)

I noticed that the vertical alignment of the clear indicator was a bit 'off', because of the outer wrapping div - pushing a tiny styling tweak to line it up better with the other text and the dropdown indicator.

Before:
Before

After:
After

**Pull request merged**